test/envtest/simple_deploy_test.go

```diff
@@ -227,6 +211,24 @@ var _ = Describe("operator processing a simple spec", Ordered, func() {
 		}, 20*time.Second*T_SCALE, EVENTUALLY_CHECK_PERIOD).Should(BeTrue())
 	})
 
+	It("should not create an ingress if disabled in CR", func() {
+		ingressKey = types.NamespacedName{Namespace: registry.Namespace, Name: registry.Name + "-ingress"}
+		Eventually(func() error {
+			Expect(s.k8sClient.Get(s.ctx, registryKey, registry)).To(Succeed())
+			registry.Spec.Deployment.ManagedResources.DisableIngress = true
+			return s.k8sClient.Update(s.ctx, registry)
+		}, 10*time.Second*T_SCALE, EVENTUALLY_CHECK_PERIOD).Should(Succeed())
+		Eventually(func() bool {
```

Comment:
I think that this step might succeed even if the Ingress got eventually created.
I propose the following:

- Deploy registry with the ingress
- Disable it
- Verify that it got removed

Reply:
Thanks for review @andreaTP.

- registry with ingress is deployed & tested at [Line 97](https://github.com/Apicurio/apicurio-registry-operator/pull/218/files#diff-f9a46e72d47305d229fe5389d38be9f4be0faa0b65b1e28e17921daf4dea243cR97).

Please correct me, if i'm wrong.

Reply:
What I'm afraid of, is the fact that this test might pass even if it actually creates the ingress.
Since there is some latency between the time Registry is reconciled and the Ingress effectively comes up it might be enough to make the test incorrectly pass.

The suggestion here would be:
in the code break the functionality (e.g. comment out the switch so that the ingress is always created) and run the test a few times, if everything looks fine I'm ok with it as it might already have enough guardrails.

Reply:
Thanks, got the new point of view. this hadn't come up in my mind.

Reply:
hey @andreaTP , i 've run the test as you suggested and everything looks fine from my side.